src/typed_array.rs

```diff
@@ -0,0 +1,360 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::{sys, JSException, JSObject, JSTypedArray, JSTypedArrayType, JSValue};
+use std::{ptr, slice};
+
+impl JSTypedArray {
+    /// Create a new [`Self`] from its raw pointer directly.
+    ///
+    /// # Safety
+    ///
+    /// Ensure `raw` is valid.
+    pub(crate) unsafe fn from_raw(ctx: sys::JSContextRef, raw: sys::JSObjectRef) -> Self {
+        Self { ctx, raw }
+    }
+
+    /// Returns a value of type [`JSTypedArrayType`] that identifies value's
+    /// Typed Array type, or `JSTypedArrayType::None` if the value is not a Typed Array
+    /// object.
+    ///
+    /// ```rust
+    /// # use javascriptcore::*;
+    /// let ctx = JSContext::default();
+    /// let array = evaluate_script(
+    ///     &ctx,
+    ///     "new Uint8Array([1, 2, 3, 4, 5])",
+    ///     None,
+    ///     "foo.js",
+    ///     1,
+    /// )
```

Comment:
Formatting? Also similar code below…

Reply:
It's not the same code exactly, we aren't asserting the same thing :-).

Code formatted.